syntax="proto3";

package messages;

import "google/protobuf/timestamp.proto";
import "google/protobuf/struct.proto";


message TrainingStatus {
  string task_id = 1;
  string phase = 2;
  float progress = 3;
  string description = 4;
  bool is_completed = 5;
  google.protobuf.Timestamp updated_at = 6;
}


message WorkerStatus {
  // UUID
  string id = 1;
  string task_type = 2;
  string version = 3;
  bool healthy = 4;
  bool has_task = 5;
  optional google.protobuf.Timestamp joined_at = 6;
  google.protobuf.Timestamp created_at = 7;
  google.protobuf.Timestamp reported_at = 8;
}

message WorkerData {
  string host = 1;
  uint32 port = 2;
  string task_type = 3;
  string version = 4;
  google.protobuf.Struct options = 5;
}
